src/helpers.js

```diff
@@ -68,12 +68,15 @@ export const getExistingCacheEntriesForCurrentBranch = async (token, prefix) =>
   const actionsCaches = new Set(defaultActionsCaches ?? []);
 
   const currentBranchRef = getCurrentBranchRef();
-  if (currentBranchRef !== defaultBranchRef) {
-    const refActionsCaches = await getExistingCacheEntries(token, prefix, currentBranchRef);
-    core.info(`Found ${refActionsCaches.length} caches for current branch ref '${currentBranchRef}'`);
 
-    actionsCaches.add(...refActionsCaches);
+  if (currentBranchRef === defaultBranchRef) {
+    return actionsCaches;
   }
 
+  const refActionsCaches = await getExistingCacheEntries(token, prefix, currentBranchRef);
+  core.info(`Found ${refActionsCaches.length} caches for current branch ref '${currentBranchRef}'`);
+
+  refActionsCaches.filter((key) => !!key).forEach((cacheKey) => actionsCaches.add(cacheKey));
```

Comment:
I dont think it's possible to get null from the cache list endpoint itself (was just that spreading empty into .add left its param undefined)

May hide issues if the cache list did for some reason though (e.g. due to github instability), but unlikely to matter either way